src/source-dpdk.c

```diff
@@ -546,9 +582,19 @@ static TmEcode ReceiveDPDKLoop(ThreadVars *tv, void *data, void *slot)
                 rte_pktmbuf_free(ptv->received_mbufs[i]);
                 continue;
             }
-            DPDKSegmentedMbufWarning(ptv->received_mbufs[i]);
-            PacketSetData(p, rte_pktmbuf_mtod(p->dpdk_v.mbuf, uint8_t *),
-                    rte_pktmbuf_pkt_len(p->dpdk_v.mbuf));
+
+            if (likely(rte_pktmbuf_is_contiguous(p->dpdk_v.mbuf))) {
+                PacketSetData(p, rte_pktmbuf_mtod(p->dpdk_v.mbuf, uint8_t *),
+                        rte_pktmbuf_pkt_len(p->dpdk_v.mbuf));
+            } else {
+                /* Slow path: segmented mbuf, needs linearization or copy */
+                DPDKSegmentedMbufWarning();
+                if (DPDKSegmentedMbufHandle(p, p->dpdk_v.mbuf) != 0) {
```

Comment:
I would suggest continuing with the processing of at least the part of the packet that we can linearize, and not giving up on it.